external-crates/move/crates/move-compiler/src/typing/ast.rs

```diff
@@ -209,7 +209,17 @@ pub enum UnannotatedExp_ {
         body: Box<Exp>,
     },
     NamedBlock(BlockLabel, Sequence),
-    Block(Sequence),
+    Block(
+        Sequence,
+        /* if representing inlined macro */
+        Option<(
+            ModuleIdent,
+            FunctionName,
+            /* method name */ Option<Name>,
+            /* type params */ Vec<Type>,
+            /* num by-value-args */ usize,
+        )>,
```

Comment:
Two nits, 
1. put it before the `Sequence`
2. Let's just make an `enum`, that way we don't need comments on what all the elements in this tuple do. Plus, it gives us a spot to add `Lambda` if we need it later

Reply:
I went for @cgswords suggestion and introduced a new `ExpandedMacro` node (only at typing) that is produced only if the compiler is in the IDE mode.